sdk/src/client/secret/ledger_nano.rs

```diff
@@ -244,39 +237,29 @@ impl SecretManage for LedgerSecretManager {
         let input_len = prepared_transaction.inputs_data.len();
 
         for input in &prepared_transaction.inputs_data {
-            let bip32_indices = match &input.chain {
-                Some(chain) => {
-                    chain
-                        .segments()
-                        .iter()
-                        // XXX: "ser32(i)". RTFSC: [crypto::keys::slip10::Segment::from_u32()]
-                        .map(|seg| u32::from_be_bytes(seg.bs()))
-                        .collect::<Vec<_>>()
-                }
-                None => return Err(Error::MissingBip32Chain)?,
-            };
+            let chain = input.chain.ok_or(Error::MissingBip32Chain)?;
 
             // coin_type and account_index should be the same in each output
-            if (coin_type.is_some() && coin_type != Some(bip32_indices[1]))
-                || (account_index.is_some() && account_index != Some(bip32_indices[2]))
+            if (coin_type.is_some() && coin_type != Some(chain.coin_type))
+                || (account_index.is_some() && account_index != Some(chain.account))
```

Comment:
maybe use `matches!` instead?

Reply:
I doubt it would be shorter

Reply:
i know it isn't ... can be closed if you don't want to change.

Reply:
Close your own comments!!!